### ISSUE TITLE
imgtool 1.9.0 Generates Invalid Hex File

### ISSUE THREAD

**Issue by legoabram** [zephyrproject-rtos/zephyr#64098]
**Describe the bug**
When using `imgtool` 1.9.0, the generated `zephyr.signed.confirmed.hex` file places the `image_ok` indicator in the wrong place when using the  `CONFIG_MCUBOOT_GENERATE_CONFIRMED_IMAGE` configuration. The bin is built correctly. Upgrading to `imgtool` 1.10.0 resolves the issue.

**To Reproduce**
1. Force install the broken version `pip install 'imgtool==1.9.0'`
2. Set `CONFIG_MCUBOOT_GENERATE_CONFIRMED_IMAGE=y`
3. Run Build
4. See incorrect location for `01` at `CF20`
```
...
:02000004080FE3
:10CF200001FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF0F
:10CF3000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF01
:10FFF00020002DE15D29410B8D77679C110F1F8A31
:00000001FF
```

**Expected behavior**
5. Force install latest `pip install 'imgtool>1.9.0'` or `pip install 'imgtool==1.10.0'`
6. Rerun build
7. See correct location for `01` at `FFC0`
```
...
:02000004080FE3
:10FFC00001FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF3F
:10FFD000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF31
:10FFF00020002DE15D29410B8D77679C110F1F8A31
:00000001FF
```

**Impact**
While not a showstopper, I spent way too long trying to figure out why OTA wasn't working since I was using the `.hex` files to program the chip in my debugger. It was causing a lockup where the image would be reverted on the next run, and is not recoverable.

**Logs and console output**
N/A

**Environment (please complete the following information):**
 - OS:  Windows 11
 - Toolchain: GCC arm-none-eabi 12.2.1
 - Commit: Zephyr 3.5.0-rc1

**Comment by bbilas** [comment]
It looks the same as https://github.com/zephyrproject-rtos/zephyr/issues/46414

**Comment by legoabram** [comment]
> It looks the same as #46414

Yep. That's when it was fixed in the imgtool branch. But the fix (https://github.com/mcu-tools/mcuboot/pull/1445) was merged after the Mar 2, 2022 release of 1.9.0.
